src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -846,6 +854,33 @@ void primitive_inst::do_runtime_skip_gather() {
     set_can_be_optimized(true);
 }
 
+void primitive_inst::do_runtime_skip_assign() {
+    OV_ITT_SCOPED_TASK(ov::intel_gpu::itt::domains::intel_gpu_plugin, openvino::itt::handle("do_runtime_skip_assign: " + id()));
+
+    if (can_be_optimized())
+        return;
+
+    if (_impl_params->has_fused_primitives())
+        return;
+
+    if (get_node().is_type<concatenation>()) {
+        for (auto u : get_user_insts()) {
+            if (u->get_node().is_type<assign>()) {
+                GPU_DEBUG_TRACE_DETAIL << "[do runtime skip reorder] update shape for user " << u->id() << std::endl;
+                u->update_shape();
+                u->update_shape_done_by_other = true;
+
+                if (u->_impl_params->get_input_layout() == u->_impl_params->get_output_layout()) {
```

Comment:
Could you clarify when this is not true? I think assign don't modify layout input anyhow